lib/orderbook/TradingPair.ts

```diff
@@ -317,27 +323,36 @@ class TradingPair {
           const matchedMakerOrder = TradingPair.splitOrderByQuantity(makerOrder, matchingQuantity);
           this.logger.debug(`reduced order ${makerOrder.id} by ${matchingQuantity} quantity while matching order ${takerOrder.id}`);
           matches.push({ maker: matchedMakerOrder, taker: remainingOrder });
-        } else if (makerFullyMatched) {
+        } else if (makerFullyMatched || makerAvailableQuantityFullyMatched) {
           // maker order quantity is not sufficient. taker order will split
           const matchedTakerOrder = TradingPair.splitOrderByQuantity(remainingOrder, matchingQuantity);
-          matches.push({ maker: makerOrder, taker: matchedTakerOrder });
+          matches.push({ maker: makerAvailableQuantityOrder, taker: matchedTakerOrder });
         } else {
-          assert(false, 'matchingQuantity should not be lower than both orders quantity values');
+          assert(false, 'matchingQuantity should not be lower than both orders available quantity values');
         }
 
         if (remainingFullyMatched) {
           remainingOrder = undefined;
         }
+
         if (makerFullyMatched) {
           // maker order is fully matched, so remove it from the queue and map
           assert(queue.poll() === makerOrder);
           const map = this.getOrderMap(makerOrder)!;
           map.delete(makerOrder.id);
           this.logger.debug(`removed order ${makerOrder.id} while matching order ${takerOrder.id}`);
+        } else if (makerAvailableQuantityFullyMatched) {
+          assert(isOwnOrder(makerOrder));
```

Comment:
What's the point of this assert? Isn't it possible that the maker order is a peer order, but still is fully matched?

Reply:
only ownOrder can be `makerAvailableQuantityFullyMatched` but not `makerFullyMatched`. So it's just for checking our own code.

Reply:
Oh right, I overlooked that this `else if` block will be skipped if `makerFullyMatched` is true. You could maybe add a comment here explaining the assert, but otherwise this makes sense.